snippets/2_tricky_strings.py

```diff
@@ -0,0 +1,31 @@
+# 1
+assert id("some_string") == id("some" + "_" + "string")
+assert id("some_string") == id("some_string")
```

Comment:
And let's not use asserts, instead I find the below format to be easily comprehensible

```py
id("some_string") == id("some" + "_" + "string")
# returns True

id("some_string") == id("some_string")
# returns True
```

I get that asserts are a Pythonically better way to validate the snippets, but they may throw beginners off.

Reply:
You're right, but `assert`s help to test snippets on various `python` versions. If expression return `False`, script won't fail and exceptions are not raised. I suggest adding a note at the beginning of `README` with examples of how `asserts` work.
Another reason to stick to `asserts` is auto checks and testing new snippets. We don't want unexpected behavior on specific `python` versions, and we definitely don't want to test each snippet manually.

Reply:
In that case, let's add comments with every assert statement as to why we're doing it. For example, in this case

```py
# id of the string object and concatenated string object with same characters is the same
assert id("some_string") == id("some" + "_" + "string")

# id of two string objects with same characters is same
assert id("some_string") == id("some_string")
```

Something like that? I know it's bit of extra work, but we shouldn't compromise on readability, that's what makes this project well-received in community

Reply:
We can even have an LLM generate these texts later if you like

Reply:
> In that case, let's add comments with every assert statement as to why we're doing it.

Sounds good except one issue - translation. We surely don't want to duplicate code just because of translated comments. That is why it is difficult to find some suitable solution. I will think about this issue and suggest some options in the meantime.

Reply:
Here is my proposed option. Code in code blocks is embedded by [this](https://github.com/tokusumi/markdown-embed-code/blob/main/markdown_embed_code/__init__.py) script which we can copy and adapt to our needs. Useful comments are outside of code block so they can be easily translated.
[example.md](https://github.com/user-attachments/files/17471760/example.md)

Reply:
sounds good, let's go with this approach @nifadyev